chia/full_node/mempool_manager.py

```diff
@@ -484,30 +513,52 @@ async def new_peak(self, new_peak: Optional[BlockRecord]) -> List[Tuple[SpendBun
             return []
         assert new_peak.timestamp is not None
 
+        use_optimization: bool = self.peak is not None and new_peak.prev_transaction_block_hash == self.peak.header_hash
```

Comment:
is this needed, I though changes are tracked across batch updates?

Reply:
Well after sync we call this method, but we dont have all of the coin changes since the start of the sync. We could instead clear the mepool after the sync.